Match pod resource metrics to pod not container in node resource metrics API tests

hasheddan opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

/kind failing-test

**What this PR does / why we need it**:

Update node resource metrics e2e tests to match pod metrics to pod
instead of containers. Also runs ResourceMetricsAPI test as part of
NodeConformance

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #96568


**Special notes for your reviewer**:

Follow-up to https://github.com/kubernetes/kubernetes/pull/95839

/cc @egernst @smarterclayton 
/sig node

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

hasheddan committed:
Match pod resource metrics to pod not container

Update node resource metrics e2e tests to match pod metrics to pod
instread of containers.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

hasheddan commented:
/milestone v1.20

egernst commented on test/e2e_node/resource_metrics_test.go:
```diff
@@ -88,12 +88,12 @@ var _ = framework.KubeDescribe("ResourceMetricsAPI [NodeFeature:ResourceMetrics]
 				}),
 
-				"pod_memory_working_set_bytes": gstruct.MatchAllElements(containerID, gstruct.Elements{
+				"pod_memory_working_set_bytes": gstruct.MatchAllElements(podID, gstruct.Elements{
```
mea culpa, good catch @hasheddan

egernst reviewed (approved):
Looks good to me, though I don't have background to weigh in on the node-conformance part.

hasheddan commented:
> Looks good to me, though I don't have background to weigh in on the node-conformance part.

Yeah I will defer to @derekwaynecarr here. Since it is a "feature" I think that we will want to remove it from `[NodeConformance]`.

egernst commented:
/assign @derekwaynecarr

hasheddan commented:
Ah need to remove the `MatchAll`. Will send update

hasheddan committed:
Relax matching on pod_memory_working_set_bytes metrics

Relaxes matching of pod_memory_working_set_bytes metric so that we won't
error due to presence of other pods.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

sjenning commented:
Yes, if you could break out the fix from the promotion to NodeConformance. Adding NodeConformance should be a separate thing.

hasheddan commented:
@sjenning done :+1:

sjenning commented:
/approve
/lgtm
/triage accepted
/priority important-soon
